corehq/apps/commtrack/models.py

```diff
@@ -410,13 +410,14 @@ def get_display_config(cls):
 
 class ActiveManager(models.Manager):
     """
-    Filter any object that is associated to an archived product.
+    Filter any object that is associated to an archived product, or deleted location.
     """
 
     def get_queryset(self):
         return super(ActiveManager, self).get_query_set() \
             .exclude(sql_product__is_archived=True) \
-            .exclude(sql_location__is_archived=True)
+            .exclude(sql_location__is_archived=True) \
+            .exclude(sql_location=None)
```

Comment:
I would check to see that no existing StockStates have `sql_location == None`, as this might change behavior unexpectedly.

You might also be able to just delete these StockStates.  @sheelio What do you think?  In my opinion, if everything is deleted reversibly, then how is that different than archiving?